velox/expression/CastExpr.cpp

```diff
@@ -744,6 +749,54 @@ void CastExpr::applyPeeled(
   }
 }
 
+VectorPtr CastExpr::applyTimestampToVarcharCast(
+    const TypePtr& toType,
+    const SelectivityVector& rows,
+    exec::EvalCtx& context,
+    const BaseVector& input) {
+  VectorPtr result;
+  context.ensureWritable(rows, toType, result);
+  (*result).clearNulls(rows);
+  auto flatResult = result->asFlatVector<StringView>();
+  const auto simpleInput = input.as<SimpleVector<Timestamp>>();
+
+  const auto& options = hooks_->timestampToStringOptions();
+  const uint32_t rowSize = getMaxStringLength(options);
+
+  Buffer* buffer =
+      flatResult->getBufferWithSpace(rows.countSelected() * rowSize);
+  char* rawBuffer = buffer->asMutable<char>() + buffer->size();
+
+  if (options.timeZone) {
+    applyToSelectedNoThrowLocal(context, rows, result, [&](vector_size_t row) {
+      // Adjust input timestamp according the session timezone.
+      Timestamp inputValue(simpleInput->valueAt(row));
+      inputValue.toTimezone(*(options.timeZone));
+      const auto stringView =
+          Timestamp::tsToStringView(inputValue, options, rawBuffer);
+      flatResult->setNoCopy(row, stringView);
+      // If string view is inline, corresponding bytes on the raw string
+      // buffer are not needed.
+      if (!stringView.isInline()) {
+        rawBuffer += stringView.size();
+      }
+    });
+  } else {
+    applyToSelectedNoThrowLocal(context, rows, result, [&](vector_size_t row) {
+      const auto stringView = Timestamp::tsToStringView(
+          simpleInput->valueAt(row), options, rawBuffer);
+      flatResult->setNoCopy(row, stringView);
+      if (!stringView.isInline()) {
+        rawBuffer += stringView.size();
+      }
+    });
+  }
+
+  // Update the exact buffer size.
+  buffer->setSize(rawBuffer - buffer->asMutable<char>());
```

Comment:
The size could not be zero, because even for Timestamp(0, 0), the output is `1970-01-01 00:00:00` (Spark) or `1970-01-01 00:00:00.000` (Presto). The two cases are covered in unit tests.

Reply:
I see. Does that mean that the check on L781 is always true?

```
if (!stringView.isInline()) {
```

If so, perhaps, change it to VELOX_DCHECK and remove or update the comment.

```
VELOX_DCHECK(!stringView.isInline())
rawBuffer += stringView.size();
```